dask/optimization.py

```diff
@@ -411,16 +411,19 @@ def _enforce_max_key_limit(key_name):
         return (_enforce_max_key_limit(concatenated_name),) + first_key[1:]
 
 
+_use_config = object()
```

Comment:
In other places we use `"__no_default__"` for this. Since tooling like sphinx, ipython, etc... will repr the defaults in the signature, having something more human-readable can be nice.

Alternatively, in `skein` I use something like:

```
# overriding `__reduce__` lets the option be properly picklable as a singleton as well.
default = type('default', (object,),
                dict.fromkeys(['__repr__', '__reduce__'],
                              lambda s: 'default'))()
```

for the same purpose.

Either is fine, but I'd prefer something other than a raw `object()`.

Reply:
I'll change it to a PEP-compliant pattern https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions

Reply:
Done, see if you like it now

Reply:
Sorry, the issue wasn't around typing or any of that, but more how it displays to the user in ipython/sphinx/etc.... Using an enum for this is fine, provided it has a descriptive repr (`UseConfig.token` does not).

We don't currently use python type annotations in dask, adding them is a separate issue and shouldn't be done here (IMO).

Perhaps

```python
In [12]: class Default(enum.Enum):
    ...:     value = ""
    ...:     def __repr__(self):
    ...:         return "<default>"
    ...:
    ...:

In [13]: def foo(x=Default.value):
    ...:     pass
    ...:

In [14]: foo?
Signature: foo(x=<default>)
Docstring: <no docstring>
File:      ~/Code/dask/<ipython-input-13-ed0fa1715aa4>
Type:      function
```

Reply:
Modified as requested.

Sphinx HTML rendered output:

```
dask.optimization.fuse(dsk, keys=None, dependencies=None, ave_width=<default>,
max_width=<default>, max_height=<default>, max_depth_new_edges=<default>,
rename_keys=<default>, fuse_subgraphs=<default>)
```